### PR TITLE
fix(deps): use new inquirer

### PR DESCRIPTION
**Pull request by mshanemc**
> requires https://github.com/salesforcecli/cli-plugins-testkit/pull/610 for select automation

### What does this PR do?
new inquirer via sf-plugins-core

other

- test fixture for `audit:messages` instead of putting the plugin-generation command in the "unit test"
- move command execution for audit:message into a nut and expand coverage
- hide `configure` commands that aren't useful outside salesforce

### What issues does this PR fix or reference?
@W-13801582@

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,11 +5,14 @@
...
-    "@oclif/core": "^3.14.1",
+    "@inquirer/confirm": "^2.0.15",
+    "@inquirer/input": "^1.2.14",
+    "@inquirer/select": "^1.3.1",
+    "@oclif/core": "^3.16.0",
     "@octokit/rest": "^19.0.13",
     "@salesforce/core": "^6.4.4",
     "@salesforce/kit": "^3.0.15",
-    "@salesforce/sf-plugins-core": "^5.0.13",
+    "@salesforce/sf-plugins-core": "5.0.14-dev.0",
```
Comment (mdonnalley) [inline package.json:15]: `^7`

File: src/generators/plugin.ts [hunk 2]
```diff
@@ -186,11 +166,11 @@ export default class Plugin extends Generator {
     }
   }
 
-  private async getGitUsername(): Promise<string | null> {
+  private async getGitUsername(): Promise<string | undefined> {
     try {
       return await this.user.github.username();
     } catch {
-      return null;
+      return undefined;
```
Comment (mdonnalley) [inline src/generators/plugin.ts:173]: `return` is the same as `return undefined`

**Review by mdonnalley — CHANGES_REQUESTED** [review]

**Comment by mdonnalley** [comment]
QA

`sf dev generate library`
✅ `Npm scope` prompt enforces `@` prefix and defaults to `@salesforce`
⚠️ it doesn't mention anything in the initial prompt about needing to start with `@` - nice to have??
❌ fails to proceed after submitting an answer

```
Time to build a library!
? Npm Scope @salesforce
✖ An error occured while running sf:library#prompting
Error (1): Cannot set properties of undefined (setting 'scope')
```

`sf dev generate plugin`
✅ `Are you building a plugin for an internal Salesforce team` - `Y`
  - ✅ `enter the name of your plugin` enforces, valid npm package name `plugin-` prefix, and all lowercase chars
    - ⚠️ it enforces a valid npm package name but the validation error only mentions needing `plugin-` and lowercase chars
  - ✅ `description` prompt works
  - ✅ successfully generates plugin


✅ `Are you building a plugin for an internal Salesforce team` - `N`
  - ✅ `enter the name of your plugin` enforces valid npm package name
  - ✅ `description` prompt works
  - ✅ `author` prompt works
  - ✅ `code coverage` selector works
  - ✅ successfully generates plugin

`sf dev generate flag`
✅ `command` selector shows all existing commands
✅ `flag type` selector shows all flag types
  - ✅ `boolean` prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char) required
  - ✅ `custom` prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char) required, multiple
    - ❌ resulting custom flag is invalid. Needs to have a `()` at the end of `Flags.custom`, e.g. `Flags.custom({summary: 'foo'})(),`
  - ✅ `directory` prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char) required, multiple, `exists` prop
  - ✅ `duration` prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char) required, multiple
    - ❌  It does not prompt for duration's unique props (like `durationUnit`)
  - ✅ `file` prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char) required, multiple, `exists` prop
  - ✅ `integer` prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char) required, multiple, `min`, `max`, `default` props
    - ❌ allows you to leave default unset even if you set a min or max
  - ✅ `option` prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char) required, multiple
    - ❌ resulting option flag is invalid. Needs to have a `()` at the end of `Flags.option`, e.g. `Flags.option({summary: 'foo'})(),`
    - ⚠️ does this need to prompt for what the `options` should be? That way we can set `options: ['foo', 'bar'] as const` for them
  - ✅ `optionalHub` prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char) required, multiple
  - ✅ `optionalOrg` prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char) required, multiple
  - ✅ `orgApiVersion` prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char) required, multiple
  - ✅ `requiredHub` prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char) required, multiple
  - ✅ `requiredOrg` prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char) required, multiple
  - ✅ `salesforceId` prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char) required, multiple, length (`both`, `15`, `18`, `none`), 3 char prefix (enforces length of 3)
  - ✅ `string` prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char) required, multiple
  - ✅ `url` prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char) required, multiple

**Comment by mdonnalley** [comment]
QA Round 2

`sf dev generate library`
> ⚠️ it doesn't mention anything in the initial prompt about needing to start with @ - nice to have??
🟢 Resolved

> ❌ fails to proceed after submitting an answer
🟢 Resolved

`sf dev generate plugin`
> ⚠️ it enforces a valid npm package name but the validation error only mentions needing plugin- and lowercase chars

Still a ⚠️ - non blocking though

`sf dev generate flag`
>❌ resulting custom flag is invalid. Needs to have a () at the end of Flags.custom, e.g. Flags.custom({summary: 'foo'})(),
🟢 Resolved
⚠️ `options` are not populated on new custom flag
⚠️ It seems odd to me to prompt for `options` on a `custom` flag since it's not required and there a ton of uses of the custom flag that don't involve `options`. 

>❌ It does not prompt for duration's unique props (like durationUnit)
🟢 Resolved

> ❌ allows you to leave default unset even if you set a min or max
🟢 Resolved

>❌ resulting option flag is invalid. Needs to have a () at the end of Flags.option, e.g. Flags.option({summary: 'foo'})(),
🟢 Resolved
⚠️ `options` are not populated on new custom flag

**Comment by mdonnalley** [comment]
QA Round 3

`sf dev generate flag`
✅ command selector shows all existing commands
✅ flag type selector shows all flag types
✅ boolean prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char), required
✅ custom prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char), required, multiple; resulting custom flag is valid.
✅ directory prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char), required, multiple, exists prop
✅ duration prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char), required, multiple, and custom props, default value is forced to between min and max
✅ file prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char), required, multiple, exists prop
✅ integer prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char), required, multiple, min, max, default props, default value is forced to between min and max
✅ option prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char), required, multiple, and options, resulting option flag is valid.

✅ all org flags prompt for "standard" props.
  - ✅ `n` sends user to typical prompts (name, summary, short char, required, multiple)
  - ✅ `y` sends user to name prompt
  - ⚠️ The standard definition prompts mentions `(character, summary, description, default)` and then the resulting flag doesn't have any of those props on it. This makes sense since `Flags.optionalHub()` has all those props set on it already but I'm wondering if it might be surprising to users - like maybe they'd be expecting defaults to populated for those properties?? Probably not a big deal - might just need slightly different messaging if it turns out to be confusing to people

✅ salesforceId prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char), required, multiple, length (both, 15, 18, none), 3 char prefix (enforces length of 3)
✅ string prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char), required, multiple
✅ url prompts for flag name (enforces uniqueness), summary (enforces capital letter and period), short char (validates single char), required, multiple

**Comment by mshanemc** [comment]
> * ⚠️ The standard definition prompts mentions `(character, summary, description, default)` and then the resulting flag doesn't have any of those props on it. This makes sense since `Flags.optionalHub()` has all those props set on it already but I'm wondering if it might be surprising to users - like maybe they'd be expecting defaults to populated for those properties?? Probably not a big deal - might just need slightly different messaging if it turns out to be confusing to people

Yeah, I chose "don't put all those on there if they're not needed" vs. duplicating what's on the prebuilt flag.  That's what our code usually does.  That also gives the ability to pick up message improvements like https://github.com/salesforcecli/sf-plugins-core/pull/455/files (they wouldn't have gotten that if we had copied those into their flag)

**Comment by mdonnalley** [comment]
> Yeah, I chose "don't put all those on there if they're not needed" vs. duplicating what's on the prebuilt flag. That's what our code usually does. That also gives the ability to pick up message improvements like https://github.com/salesforcecli/sf-plugins-core/pull/455/files (they wouldn't have gotten that if we had copied those into their flag)

To be clear, I wasn't advocating for adding the props when they're not needed - just that it might be surprising to developers who aren't familiar with how the flags work

**Review by mdonnalley — APPROVED** [review]

**Pull request merged**